### ISSUE TITLE
Formulas in np.array() support

### ISSUE THREAD

**Issue by airguru** [cunnane/xloil#25]
Would it be possible to support bulk editing of cell formulas using the range object? It is very slow to change large amounts of formulas individually. 

It could very much be converted to 2d numpy string / dtype('O') array and back. 

Alternatively, is there an equivalent of stopping and re-enabling workbook calculation and redraw for the duration of the macro just as it could be done in pure VBA? 

Thanks in advance for any ideas!

**Comment by cunnane** [comment]
It would be possible, but it's not supported by Excel's API, so xlOil would only be able to do this in a loop behind-the-scenes.

You can pause Excel's processing with the following, very similar to what you'd write in VBA:

	from win32com.client import constants as c
	try:
		xlo.app().Calculation = c.xlCalculationManual
		xlo.app().ScreenUpdating = False
		with xloil.EventsPaused() as p:
			# write formulae
	finally:
		xlo.app().Calculation = c.xlCalculationAutomatic
		xlo.app().ScreenUpdating = True

It look like extending the `EventsPaused` context manager would be a sensible change to streamline this.

**Comment by airguru** [comment]
I have succesfully tried this, but to my surprise, I need to use the `to_com()` method:
```
xlo.app().to_com().Calculation = c.xlCalculationManual
```

Without that, the properties of the `app()` cannot be assigned to.
```
-> xlo.app()
<xloil_core.Application object at 0x00000000511F7070>
-> xlo.app().Calculation
-4105
-> xlo.app().Calculation = c.xlCalculationManual
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'xloil_core.Application' object has no attribute 'Calculation'
```

Maybe I am using wrong com library? I have `pywin32==305`

**Comment by cunnane** [comment]
No there was an issue which has been fixed in xlOil 0.17, so the expected syntax should now work.

I also added [PauseExcel](https://xloil.readthedocs.io/en/stable/xlOil_Python/ModuleReference.html#xloil.PauseExcel), a context manager which takes care of running the above COM commands.
